crates/telio-starcast/src/multicaster.rs

```diff
@@ -0,0 +1,281 @@
+use async_trait::async_trait;
+use futures::future::join_all;
+use pnet_packet::ipv4::{checksum, MutableIpv4Packet};
+use pnet_packet::udp::MutableUdpPacket;
+use std::collections::hash_map::DefaultHasher;
+use std::collections::HashMap;
+use std::hash::{Hash, Hasher};
+use std::net::{IpAddr, Ipv4Addr};
+use std::{net::SocketAddr, sync::Arc};
+use telio_sockets::SocketPool;
+use telio_task::io::Chan;
+use telio_task::{Runtime, RuntimeExt, Task, WaitResponse};
+use telio_utils::{telio_log_debug, telio_log_error, telio_log_warn};
+use telio_wg::uapi::Interface;
+use tokio::net::UdpSocket;
+use tokio::sync::RwLock;
+
+#[cfg(feature = "test-util")]
+const MULTICAST_TRANSPORT_PORT: u16 = 0;
+#[cfg(not(feature = "test-util"))]
+const MULTICAST_TRANSPORT_PORT: u16 = 13569;
+const MAX_PACKET_SIZE: usize = 2048;
+const IPV4_HEADER: usize = 20; // bytes
+const UDP_DEST_OFFSET: usize = 2;
+
+/// Multicaster task that resends locally generated
+/// multicast traffic to other peers
+pub struct Multicaster {
+    wg: Arc<Interface>,
```

Comment:
So here we need to decide witch approach we take, theare are two:
- Having Arc<dyn Wg>, witch we query then we need to consult who is allive, this would meed we alwayas have an up to date state to who we need to send messages.
- Having some cached values of nodes, that is updated by wg_controller like vpeer does: https://github.com/NordSecurity/libtelio/blob/f1687494b60f8c0c6236bf607b89156ace20eb0d/crates/telio-starcast/src/multicast_peer.rs#L300

As it currently stands its broken on both fronts :D